builder/config.go

```diff
@@ -114,13 +115,49 @@ func getWarningsForObsoleteFields(reader io.Reader) ([]string, error) {
 }
 
 // parseConfig reads a builder configuration from reader and resolves relative buildpack paths using `relativeToDir`
-func parseConfig(reader io.Reader, relativeToDir string) (Config, error) {
+func parseConfig(reader io.Reader, relativeToDir, path string) (Config, error) {
 	var builderConfig Config
-	if _, err := toml.DecodeReader(reader, &builderConfig); err != nil {
+	tomlMetadata, err := toml.DecodeReader(reader, &builderConfig)
+	if err != nil {
 		return Config{}, errors.Wrap(err, "decoding toml contents")
 	}
 
+	undecodedKeys := tomlMetadata.Undecoded()
+	if len(undecodedKeys) > 0 {
+		unusedKeys := map[string]interface{}{}
+		for _, key := range undecodedKeys {
+			keyName := key.String()
+
+			parent := strings.Split(keyName, ".")[0]
+
+			if _, ok := unusedKeys[parent]; !ok {
+				unusedKeys[keyName] = nil
+			}
+		}
+
+		var errorKeys []string
+		for errorKey := range unusedKeys {
+			errorKeys = append(errorKeys, style.Symbol(errorKey))
+		}
+
+		pluralizedElement := "element"
+		if len(errorKeys) > 1 {
+			pluralizedElement += "s"
+		}
+
+		return Config{}, errors.Errorf("unknown configuration %s %s in %s",
+			pluralizedElement,
+			strings.Join(errorKeys, ", "),
+			style.Symbol(path),
+		)
+	}
+
+	fmt.Println(builderConfig.Buildpacks.Buildpacks())
 	for i, bp := range builderConfig.Buildpacks.Buildpacks() {
+		if bp.URI == "" {
+			return Config{}, errors.Errorf("missing %s configuration", style.Symbol("buildpack.uri"))
+		}
+
```

Comment:
This check is actually unnecessary since the call to `.Buildpacks()` scopes the list to bp's with non-empty URI

Reply:
got it!